### PR TITLE
feat: add beta anchored region component

### PR DESCRIPTION
**Pull request by scomea**
# Description
A pr to clean up and publish beta version of anchored region and associated api's.

## Motivation & context
Make anchored region a usuable fast component.

## Issue type checklist
- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**
- [x] I have added a new component
- [x] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist
- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [x] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

File: packages/web-components/fast-components/src/anchored-region/anchored-region.definition.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                //     description: "The horizontal-positioning-mode attribute",
+                //     type: DataType.string,
+                //     values: [
+                //         { name: AxisPositioningMode.uncontrolled },
+                //         { name: AxisPositioningMode.locktodefault },
+                //         { name: AxisPositioningMode.dynamic },
+                //     ],
+                //     default: true,
+                //     required: false,
+                // },
```
Comment (chrisdholt) [inline packages/web-components/fast-components/src/anchored-region/anchored-region.definition.ts:37]: Should this be removed?

File: packages/web-components/fast-foundation/src/anchored-region/anchored-region.ts [hunk 2]
```diff
@@ -18,13 +18,53 @@ declare global {
...
+ *
+ * @beta
+ */
+export enum VerticalPosition {
+    top = "top",
+    bottom = "bottom",
+    unset = "unset",
+}
+
+interface Dimension {
```
Comment (chrisdholt) [inline packages/web-components/fast-foundation/src/anchored-region/anchored-region.ts:67]: Should this be exported?

Comment (chrisdholt) [reply packages/web-components/fast-foundation/src/anchored-region/anchored-region.ts:67]: Also, should we include typedoc info here? If it's exported - is it public or internal?

File: packages/web-components/fast-components/src/anchored-region/anchored-region.definition.ts (hunk 1 above)
Comment (scomea) [reply packages/web-components/fast-components/src/anchored-region/anchored-region.definition.ts]: actually needed to finish filling in the attributes.

File: packages/web-components/fast-foundation/src/anchored-region/anchored-region.ts [hunk 3]
```diff
@@ -18,13 +18,56 @@ declare global {
...
+ */
+export enum VerticalPosition {
+    top = "top",
+    bottom = "bottom",
+    unset = "unset",
+}
+
+/**
+ * used internally to track dimensions of the region
+ */
```
Comment (chrisdholt) [inline packages/web-components/fast-foundation/src/anchored-region/anchored-region.ts:69]:
```suggestion
 * @internal
 */
```

File: packages/web-components/fast-foundation/src/anchored-region/anchored-region.ts [hunk 4]
```diff
@@ -18,13 +18,56 @@ declare global {
     }
 }
 
-export type AxisPositioningMode = "uncontrolled" | "locktodefault" | "dynamic";
-export type AxisScalingMode = "anchor" | "fill" | "content";
-export type GbcrUsage = "always" | "never" | "default";
-export type HorizontalPosition = "start" | "end" | "left" | "right" | "unset";
-export type VerticalPosition = "top" | "bottom" | "unset";
```
Comment (chrisdholt) [inline packages/web-components/fast-foundation/src/anchored-region/anchored-region.ts:25]: Is there a reason for making the change to having these be enums? Enums don't compress whatsoever and types provide type safety. I'd prefer to keep these as types unless there is an explicit reason to change it over to an enum.

Comment (scomea) [reply packages/web-components/fast-foundation/src/anchored-region/anchored-region.ts]: types good, enum bad...converted to types

File: packages/web-components/fast-foundation/src/anchored-region/anchored-region.ts (hunk 2 above)
Comment (scomea) [reply packages/web-components/fast-foundation/src/anchored-region/anchored-region.ts:52]: marked as internal

File: packages/web-components/fast-foundation/src/anchored-region/anchored-region.ts (hunk 4 above)
Comment (chrisdholt) [reply packages/web-components/fast-foundation/src/anchored-region/anchored-region.ts]: made me laugh during a meeting 😆

**Review by chrisdholt — APPROVED** [review]

**Review by nicholasrice — APPROVED** [review]

**Pull request merged**